src/generator/AutoRest.CSharp.Azure.Fluent/ClientModelExtensions.cs

```diff
@@ -1,29 +1,32 @@
-﻿using AutoRest.Core.ClientModel;
-using AutoRest.Extensions.Azure;
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// 
+
 using System.Text.RegularExpressions;
+using AutoRest.Core.Model;
 
 namespace AutoRest.CSharp.Azure.Fluent
 {
     public static class ClientModelExtensions
     {
-        public static bool IsResource(this IType type)
+        public static bool IsResource(this IModelType type)
         {
-            CompositeType compositeType = type as CompositeType;
-            return compositeType != null && 
-                (type.Name == "Resource" || type.Name == "SubResource" ||
-                 type.Name == "Microsoft.Rest.Azure.Resource" || type.Name == "Microsoft.Rest.Azure.SubResource");
+            var compositeType = type as CompositeType;
+            return (compositeType != null) &&
+                   ((type.Name == "Resource") || (type.Name == "SubResource") ||
+                    (type.Name == "Microsoft.Rest.Azure.Resource") || (type.Name == "Microsoft.Rest.Azure.SubResource"));
```

Comment:
If this was a mainstream generator,  I'd be right there with you. At this point it's a big prototype, and I'm not sure that it's ready to even bother yet. But good catch.